library/ssl_tls13_client.c

```diff
@@ -907,16 +959,17 @@ int mbedtls_ssl_create_binder( mbedtls_ssl_context *ssl, unsigned char *psk, siz
     else
 #endif
 #if defined(MBEDTLS_SHA512_C)
-        if( suite_info->mac == MBEDTLS_MD_SHA384 )
-        {
-            mbedtls_sha512_free( &sha512 );
-        }
-        else
+    if( suite_info->mac == MBEDTLS_MD_SHA384 ||
+        suite_info->mac == MBEDTLS_MD_SHA512 )
```

Comment:
Why was this change made? It doesn't seem to fall into the pattern of replacing deprecated functions by their new variants?

Reply:
Since we're using the same context in the case of both MBEDTLS_MD_SHA384 and MBEDTLS_MD_SHA512, we should free it in any of those cases.
You're right, it's orthogonal to the PR's objective. I guess I fixed without noticing